backend/manager/modules/bll/src/main/java/org/ovirt/engine/core/bll/storage/lsm/LiveMigrateDiskCommand.java

```diff
@@ -327,8 +327,7 @@ private void removeImage(Guid storageDomainId, Guid imageGroupId, Guid imageId,
     }
 
     private boolean isRemoveAutoGeneratedSnapshotRequired() {
-        boolean removeSnapshotRequired = getParameters().getLiveDiskMigrateStage() == LiveDiskMigrateStage.LIVE_MIGRATE_DISK_EXEC_COMPLETED &&
-                getParameters().getLiveDiskMigrateStage() != LiveDiskMigrateStage.AUTO_GENERATED_SNAPSHOT_REMOVE_END;
+        boolean removeSnapshotRequired = getParameters().getLiveDiskMigrateStage() == LiveDiskMigrateStage.LIVE_MIGRATE_DISK_EXEC_COMPLETED;
         if (removeSnapshotRequired) {
             if (!getVm().getStatus().isQualifiedForLiveSnapshotMerge()) {
                 // If the VM is not qualified for live merge, i.e. its status is not up, the auto-generated snapshot
```

Comment:
As @ahadas mentioned in
https://github.com/oVirt/ovirt-engine/pull/302#discussion_r857318198
BZ [1369942](https://bugzilla.redhat.com/show_bug.cgi?id=1369942) was already resolved in 4.4.
@bennyz should I update this code? If so, add to line 332 another "AND" condition that cluster version is less than 4.4?

Reply:
block commit is used since 4.2, the check is not needed afaik

Reply:
yes

Reply:
please squash the last 2 commits

Reply:
Done :)